guava/src/com/google/common/base/Joiner.java

```diff
@@ -214,8 +211,7 @@ public final String join(Object[] parts) {
    * Returns a string containing the string representation of each argument, using the previously
    * configured separator between each.
    */
-  public final String join(
-      @NullableDecl Object first, @NullableDecl Object second, @org.checkerframework.checker.nullness.qual.Nullable Object... rest) {
+  public final String join(@Nullable Object first, @Nullable Object second, Object... rest) {
```

Comment:
You removed `@Nullable` from the array element type (the last `Object` on this line).  Why?